cpp/src/arrow/csv/converter.cc

```diff
@@ -44,6 +44,13 @@ Status GenericConversionError(const std::shared_ptr<DataType>& type, const uint8
   return Status::Invalid(ss.str());
 }
 
+inline bool IsWhitespace(uint8_t c) {
+  if (ARROW_PREDICT_TRUE(c > 0x20)) {
```

Comment:
Maybe `c > ' ` or even `c > ' ' && c > '\t'` here to make it more explicit? The last one should get optimized to a single comparison anyway.